client/src/sockets/receiveEvents.js

```diff
@@ -27,7 +27,7 @@ export const receiveEvents = (dispatch, socket) => {
   );
 
   Object.values(CohortEvents).forEach(event =>
-    socket.on(event, data => dispatch({ type: event, payload: data }))
+    socket.on(event, data => cohortEventsController(event, data, dispatch))
```

Comment:
If you kept the redux-like protocol, the previous call would fit without any changes.